vitest.config.shared.ts

```diff
@@ -0,0 +1,28 @@
+import * as vitest from 'vitest/config';
+
+export const base: vitest.ViteUserConfig = {
+  test: {
+    coverage: {
+      thresholds: { 100: true },
+      provider: 'istanbul',
+      include: ['src/**/*.ts'],
+    },
+    maxWorkers: process.env.CI ? 6 : undefined,
+    reporters: process.env.CI ? ['junit'] : [],
+    outputFile: process.env.CI ? 'reports/junit.xml' : undefined,
+  },
+};
+
+export function defineConfig(
+  config: vitest.ViteUserConfig
+): vitest.ViteUserConfig {
+  const { test, ...rest } = config;
+  return vitest.defineConfig({
+    ...base,
+    ...rest,
+    test: {
+      ...base.test,
+      ...test,
+    },
```

Comment:
Should we do a recursive object extension here so you can override individual fields (e.g. change coverage thresholds without also needing to specify `provider` and `include`)?

Reply:
Done.

Reply:
Just remembered we have a `mergeObjects` function in libs/basics

Reply:
We could copy it, I guess. Probably don't want to try to import it from here.

Reply:
Oh yeah I guess it doesn't make sense to import. Nevermind